ambari-server/src/main/java/org/apache/ambari/server/upgrade/UpgradeCatalog262.java

```diff
@@ -61,6 +66,29 @@ private void addHostRequestStatusColumn() throws SQLException {
 
   @Override
   protected void executePreDMLUpdates() throws AmbariException, SQLException {
+    fixDesiredStack();
+  }
+
+  /**
+   * if desired stack < current stack, set current stack as desired
+   *
+   * @throws AmbariException
+   */
+  private void fixDesiredStack() throws AmbariException {
+    AmbariManagementController ambariManagementController = injector.getInstance(AmbariManagementController.class);
+    Clusters clusters = ambariManagementController.getClusters();
+    if (clusters != null) {
+      Map<String, Cluster> clusterMap = getCheckedClusterMap(clusters);
+      if (clusterMap != null && !clusterMap.isEmpty()) {
+        for (final Cluster cluster : clusterMap.values()) {
+          StackId desiredStack = cluster.getDesiredStackVersion();
+          StackId currentStack = cluster.getCurrentStackVersion();
+          if (desiredStack.compareTo(currentStack) < 0) {
+            cluster.setDesiredStackVersion(currentStack);
+          }
```

Comment:
If desired stack is > current stack, that's also a problem. Can you change this to a !equals() call instead?

Reply:
I assume that customer is not expected to perform Ambari upgrade with a paused stack upgrade? Ok, will change the logic

Reply:
@jonathan-hurley , regarding 

    Why does express not have this problem? 

EU updates cluster desired version during upgrade and downgrade at `UpdateDesiredRepositoryAction`. This action is called only from EU upgrade packs.

Reply:
Yes, that's what I expected. However, I didn't see it in there. Can you identify where exactly it's set?

Reply:
`org/apache/ambari/server/serveraction/upgrades/UpdateDesiredRepositoryAction.java:149`
at `updateDesiredRepositoryVersion()`

```
        // move the cluster's desired stack as well
        StackId targetStackId = targetRepositoryVersion.getStackId();
        cluster.setDesiredStackVersion(targetStackId);
      }
```

and below at the same file for downgrade.
EU was not broken, this code is already at 2.6 branch

Reply:
I was looking in trunk :)

Thanks!